durablestore/dynamodb/dynamodb.go

```diff
@@ -40,23 +40,22 @@ func (ddb ddb) GetItem(ctx context.Context, persistenceID string) (*item, error)
 		TableName: aws.String(ddb.tableName),
 		Key:       key,
 	})
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch the latest state from the dynamodb: %w", err)
-	}
 
-	// Check if item exists
-	if result.Item == nil {
+	switch {
+	case err != nil:
+		return nil, fmt.Errorf("failed to fetch the state from the dynamodb: %w", err)
+	case result.Item == nil:
```

Comment:
What happened when `result` is nil?

Reply:
Good question. I've never encounter result empty here, but I'll handle the case anyway